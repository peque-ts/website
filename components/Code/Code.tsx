import hljs from 'highlight.js/lib/core';
import typescript from 'highlight.js/lib/languages/typescript';
import React, { useEffect, useRef } from 'react';

import { assertString } from '../../utils/assertions';

export const Code: React.FC = ({ children }) => {
  assertString(children);

  const ref = useRef(null);

  useEffect(() => {
    hljs.registerLanguage('typescript', typescript);

    /* istanbul ignore else */
    if (ref.current) {
      hljs.highlightElement(ref.current);
    }
  }, []);

  return (
    <pre className="custom-scrollbar">
      <code role="code" className="drop-shadow-xl rounded" ref={ref}>
        {children.trim()}
      </code>
    </pre>
  );
};
