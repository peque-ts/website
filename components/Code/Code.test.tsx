import { render } from '@testing-library/react';

import { Code } from './Code';

describe('Code', () => {
  it('should render', () => {
    const { asFragment } = render(<Code />);

    expect(asFragment()).toMatchSnapshot();
  });
});
