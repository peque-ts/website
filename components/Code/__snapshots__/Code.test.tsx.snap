// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code should render 1`] = `
<DocumentFragment>
  <pre
    class="custom-scrollbar"
  >
    <code
      class="drop-shadow-xl rounded hljs language-typescript"
      role="code"
    >
      <span
        class="hljs-variable language_"
      >
        console
      </span>
      .
      <span
        class="hljs-title function_"
      >
        log
      </span>
      (
      <span
        class="hljs-string"
      >
        'something'
      </span>
      );
    </code>
  </pre>
</DocumentFragment>
`;
